From 51d07f42e8a0521791bec7981bee5808cb9b4692 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <chrisg@marvell.com>
Date: Wed, 21 Feb 2018 19:33:42 +0200
Subject: [PATCH 1254/1345] dts: mcbin: add device tree for MacchiatoBin
 single shot board

commit  93d0292f559985992db9f2466dad57dc2d67fef4 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- MacchiatoBin single shot board doesn't have two 3310 PHY's
  connected to port 0 on each CP

Change-Id: Ifa81694bcf85f0f2ece127c37d6219f999423a40
Signed-off-by: Christine Gharzuzi <chrisg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/50872
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../bindings/arm/marvell/armada-7k-8k.txt          |    3 +
 arch/arm64/boot/dts/marvell/Makefile               |    1 +
 .../dts/marvell/armada-8040-mcbin-single-shot.dts  |   76 ++++++++++++++++++++
 3 files changed, 80 insertions(+)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-8040-mcbin-single-shot.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
index 75cdcab..eb36c581 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
@@ -77,6 +77,9 @@ Boards:
    - "marvell,armada8040-mcbin"
       when the board being used is the MACCHIATOBin community board
 
+   - "marvell,armada8040-mcbin-single-shot"
+      when the board being used is the MACCHIATOBin single shot community board
+
    - "marvell,armada8040-ocp"
       when the board being used is the OCP board
 
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index bea5212..8690026 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -33,6 +33,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-C.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-D.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-E.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin-single-shot.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-ocp.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-80x0-customer.dtb
 
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin-single-shot.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin-single-shot.dts
new file mode 100644
index 0000000..856228b
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin-single-shot.dts
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2018 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Device Tree file for MACCHIATOBin SingleShot Armada 8040 community board platform.
+ * Difference from MACCHIATOBin DoubleShot:
+ * MacchiatoBin single shot doesn't have two 3310 PHY's connected to port 0 on each CP.
+ */
+
+#include "armada-8040-mcbin.dts"
+
+/ {
+	model = "Marvell 8040 MACHIATOBin SingleShot";
+	compatible = "marvell,armada8040-mcbin-single-shot", "marvell,armada8040-mcbin",
+		     "marvell,armada8040", "marvell,armada-ap806-quad",
+		     "marvell,armada-ap806";
+
+	cpn-110-master {
+		config-space {
+		/delete-node/mdio@15b000;
+			gop {
+				emac0_0: mac0 {
+					/delete-property/phy;
+				};
+			};
+		};
+	};
+	cpn-110-slave {
+		config-space {
+			gop {
+				emac0_1: mac0 {
+					/delete-property/phy;
+				};
+			};
+		};
+	};
+};
-- 
1.7.9.5

